### PR TITLE
Complete rebuild of UI layer

### PR DESCRIPTION
**Pull request by devjayprakash**
I've rebuild the UI and added the following features

- Removed chakra UI (i think using a component library for a simple UI will make the website heavy and slow)
- Added vite with swc as build tools (it's the fastest build tool for react)
- Used react router v6
- Used tailwind css for styling
- Fully responsive (Looks great on mobile as well)
- Added search for stocks
- I've selected `stocksprint` as name of the site.. Let me know if you have a better idea.

What's lacking (i'll add these eventually)
- google analytics layer
- google sign up (btw a question : why do we even need this )  
- agreement page 
- read me file 

Screenshot
Screenshot 2024-10-06 at 12 23 32 PM
Screenshot 2024-10-06 at 12 23 21 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,83 +1,8 @@
...
-
-1. Fork the repository.
-2. Create a new branch for your feature or bugfix.
-3. Submit a pull request with detailed information about your changes.
-
-## License
-
-This project is licensed under the MIT License. For more information, see the [LICENSE](LICENSE) file.
+- [@vitejs/plugin-react](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react/README.md) uses [Babel](https://babeljs.io/) for Fast Refresh
+- [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react-swc) uses [SWC](https://swc.rs/) for Fast Refresh
```
Comment (shivamsouravjha) [inline README.md:8]: let's not change the readme of the project

**Comment by shivamsouravjha** [comment]
we're using the google signup thing to access the portfolio directly from mail.

**Comment by shivamsouravjha** [comment]
isntead of stocksprint we can use stockSight because our aim is for more indepth insights of a stock

**Comment by devjayprakash** [comment]
> we're using the google signup thing to access the portfolio directly from mail.

oh got it.. Thank makes sense

**Comment by shivamsouravjha** [comment]
there are few conflicts please resolved that rest everything is good to go

**Comment by shivamsouravjha** [comment]
can we put this as the logo https://github.com/shivamsouravjha/Stooler/blob/main/frontend/public/logo512.png ? (the one in top of window as well as the official logo)

**Comment by devjayprakash** [comment]
Sure i'll fix it .. Also we'll have to force push to to main as it's a complete rewrite. I'll add the google analytics as soon  in a separate PR

**Comment by shivamsouravjha** [comment]
Won’t be force pushing we can just shift those things as others contribution might be lost as well.


Lets try to put everything in this pr including the analytics.

**Comment by devjayprakash** [comment]
> Won’t be force pushing we can just shift those things as others contribution might be lost as well.
> 
> Lets try to put everything in this pr including the analytics.

Oh yes you are right..  sure i'll try to add other contributions to this one..

**Comment by shivamsouravjha** [comment]
let's try to wrap it up as there are few contributors who want to pick different task and it will be a mess for your if their prs are merged first and for them if yours  merged after their changed. rest ui looks great!

**Comment by shivamsouravjha** [comment]
can't merge because of the conflicts and since this is a fork I can't fix them either

**Comment by devjayprakash** [comment]
> can't merge because of the conflicts and since this is a fork I can't fix them either

As this is a complete re implementation it would not be possible to merge. You will have to pull this branch. Then push it by force to main .  Here is how you can do it.
1. Clone my repo somewhere (https://github.com/devjayprakash/stock-frontend)
2. Add your repo as another remote `git remote add shivam_origin  this_repo_link `
3. Then you can push it to this repo by doing `git push -u -f shivam_origin feature/rebuild_ui:main`

**Comment by shivamsouravjha** [comment]
have merged the same !

**Comment by shivamsouravjha** [comment]
* cap texts aren't show like 
image

* also we need to colorise the cards depending upon the cap (we can just colorise the heading to differenitate it) (visuale differences are better than textual)

**Pull request closed without merging**